file/src/main/scala/akka/stream/alpakka/file/internal/FileSizeRotatorFlow.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.file.internal
+
+import java.nio.file.Path
+
+import akka.stream.alpakka.file.internal.ChangeableFileSubscriber.{
+  ByteStringWrapper,
+  ChangePathMessage,
+  ChangeableFileSubscriberMessages
+}
+import akka.stream.stage.{GraphStage, GraphStageLogic, InHandler, OutHandler}
+import akka.stream.{Attributes, FlowShape, Inlet, Outlet}
+import akka.util.ByteString
+
+class FileSizeRotatorFlow(max: Long, f: () => Path)
```

Comment:
"If you liked it, make it final, also if it is internal, make it `private[file]`.
A better name for `f` would be good, "selectPath" for example.

If you abstract the decision when to ask for a new path should be asked for into a class, you could have just one stage and then just provide different factories for the case of always asking and or based on counting bytes.

I think you should be able to encapsulate all the logic needed in this stage rather than combining it with other stages. Alternatively use substreams and the built in akka streams file writer if you want to avoid touching the IO/NIO apis directly.